### ISSUE TITLE
Refactor use of `window.matchMedia` in header and tabs components

### ISSUE THREAD

**Issue by 36degrees** [alphagov/govuk-frontend#3445]
## What

After v5, JavaScript will only execute in browsers that support `script type="module"`. All browsers that support `script type="module"` [also support `window.matchMedia`](https://browsersl.ist/#q=supports+es6-module+and+not+supports+matchmedia), so we can simplify the JavaScript as we no longer need to guard against `window.matchMedia` being undefined.

~~At the same time we may also want to look at our inconsistent use of `addEventListener` versus `addListener` (required for Safari < 14) across the two components, and the fact that TypeScript ‘encourages’ us to miss the need for the addListener callback (see https://github.com/alphagov/govuk-frontend/pull/3439#discussion_r1151884480)~~

EDIT: We think it's unlikely we can do anything about this any time soon, other than use `addEventListener` when it's available in the tabs component, which we've done in #3765.

~~Finally, we have also hardcoded the default desktop breakpoint in our JavaScript in both components, which may for example cause issues if users override the default breakpoints. All browsers that support `script type="module"` [also support CSS variables](https://browsersl.ist/#q=supports+es6-module+and+not+supports+css-variables) so we might be able to expose the breakpoint in the CSS for each component (or on the document root) and read it out using `window.getComputedStyle().getPropertyValue`.~~

EDIT: Split out into #3764.

## Why

Once we've shipped v5, we can start to tidy up and refactor our component code to make it more maintainable, performant and easier to read.

## Who needs to work on this

Developers

## Who needs to review this

Developers

## Done when

The JavaScript for the header and tabs components:

- [x] no longer guards against `window.matchMedia` being undefined (#3723)
- [ ] ~~uses `addEventListener` and `addListener` consistently (or we've decided not to do this)~~
- [ ] ~~no longer includes hard-coded breakpoints (or we've decided not to do this)~~

**Comment by 36degrees** [comment]
It looks like at least the main part of this was solved in #3723. Based on the title we should be able to close this, but there are a couple of other things in the description that are tangentially related but not worth looking at right now.

> At the same time we may also want to look at our inconsistent use of addEventListener versus addListener (required for Safari < 14) across the two components, and the fact that TypeScript ‘encourages’ us to miss the need for the addListener callback (see https://github.com/alphagov/govuk-frontend/pull/3439#discussion_r1151884480)

I think it's going to be a number of years before we can resolve this as we basically need to wait until we no longer need to support or run JavaScript in Safari < 14. I was hoping that Babel might do something clever in order to transpile the call to `addEventListener` into something that works in older versions of Safari, but from playing around in the Babel sandbox [that doesn't seem to be the case](https://babeljs.io/repl#?browsers=safari%2012.1&build=&builtIns=usage&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBAtgRwDYwLwwO4EswBMQYB0cAhlMABYCyAprliQBQDkjcOAtNrlBQFwwAHAAZhABwAeASmZSAUHN5YIxZIRK5cAUQBuNMFAAyyqPpoAnFpRJgA5jWYAaGADMArmGBQs4GIykwAN5yMDCgkCBINIRIILYsAGIgIACEsnIAvlJAA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-2%2Cstage-3&prettier=false&targets=&version=7.22.4&externalPlugins=&assumptions=%7B%7D).

Given there's no obvious immediate fix I'm inclined to just forget this for now and leave it for 'distant future us' to tidy up?

> Finally, we have also hardcoded the default desktop breakpoint in our JavaScript in both components, which may for example cause issues if users override the default breakpoints. All browsers that support `script type="module"` [also support CSS variables](https://browsersl.ist/#q=supports+es6-module+and+not+supports+css-variables) so we might be able to expose the breakpoint in the CSS for each component (or on the document root) and read it out using `window.getComputedStyle().getPropertyValue`.

I've split this one out into its own issue as I still think it's worth doing but it was only ever tangentially related – https://github.com/alphagov/govuk-frontend/issues/3764

If we're happy with the above I think we can then update the description and close this issue?

**Comment by colinrotherham** [comment]
Shame about Babel

For the sake of consistency shall we add `.addEventListener()` to Tabs?

Probably since TypeScript 5.1 upgrade but I'm seeing the MDN's advice linked to `@deprecated` now:
https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener

It's fairly shouty about it 😬

>**Deprecated:** This feature is no longer recommended. Though some browsers might still support it, it may have already been removed from the relevant web standards, may be in the process of being dropped, or may only be kept for compatibility purposes. Avoid using it, and update existing code if possible; see the [compatibility table](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener#browser_compatibility) at the bottom of this page to guide your decision. Be aware that this feature may cease to work at any time.

**Comment by 36degrees** [comment]
With that PR merged and the other change split out into a separate issue I'm going to call this done.
